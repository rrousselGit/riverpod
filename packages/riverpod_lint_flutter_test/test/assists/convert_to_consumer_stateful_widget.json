[{"priority":31,"change":{"message":"Convert to ConsumerStatefulWidget","edits":[{"fileStamp":0,"edits":[{"offset":221,"length":0,"replacement":"@override\n  ConsumerState<Stateless> createState() => _StatelessState();\n}\n\nclass _StatelessState extends ConsumerState<Stateless> {\n"},{"offset":168,"length":15,"replacement":"ConsumerStatefulWidget"}]}],"linkedEditGroups":[]}},{"priority":31,"change":{"message":"Convert to ConsumerStatefulWidget","edits":[{"fileStamp":0,"edits":[{"offset":404,"length":0,"replacement":"@override\n  ConsumerState<StatelessWithComma> createState() => _StatelessWithCommaState();\n}\n\nclass _StatelessWithCommaState extends ConsumerState<StatelessWithComma> {\n"},{"offset":342,"length":15,"replacement":"ConsumerStatefulWidget"}]}],"linkedEditGroups":[]}},{"priority":31,"change":{"message":"Convert to ConsumerStatefulWidget","edits":[{"fileStamp":0,"edits":[{"offset":563,"length":0,"replacement":"@override\n  ConsumerState<Hook> createState() => _HookState();\n}\n\nclass _HookState extends ConsumerState<Hook> {\n"},{"offset":520,"length":10,"replacement":"ConsumerStatefulWidget"}]}],"linkedEditGroups":[]}},{"priority":31,"change":{"message":"Convert to ConsumerStatefulWidget","edits":[{"fileStamp":0,"edits":[{"offset":787,"length":23,"replacement":""},{"offset":737,"length":0,"replacement":"@override\n  ConsumerState<HookConsumer> createState() => _HookConsumerState();\n}\n\nclass _HookConsumerState extends ConsumerState<HookConsumer> {\n"},{"offset":678,"length":18,"replacement":"ConsumerStatefulWidget"}]}],"linkedEditGroups":[]}},{"priority":27,"change":{"message":"Convert to ConsumerStatefulWidget","edits":[{"fileStamp":0,"edits":[{"offset":1022,"length":15,"replacement":"ConsumerState<Stateful>"},{"offset":939,"length":15,"replacement":"ConsumerState<Stateful>"},{"offset":876,"length":14,"replacement":"ConsumerStatefulWidget"}]}],"linkedEditGroups":[]}},{"priority":27,"change":{"message":"Convert to ConsumerStatefulWidget","edits":[{"fileStamp":0,"edits":[{"offset":1389,"length":26,"replacement":"ConsumerState<ExplicitCreateState>"},{"offset":1203,"length":14,"replacement":"ConsumerStatefulWidget"}]}],"linkedEditGroups":[]}},{"priority":31,"change":{"message":"Convert to ConsumerStatefulWidget","edits":[{"fileStamp":0,"edits":[{"offset":1708,"length":19,"replacement":"ConsumerState<HookStateful>"},{"offset":1615,"length":19,"replacement":"ConsumerState<HookStateful>"},{"offset":1544,"length":18,"replacement":"ConsumerStatefulWidget"}]}],"linkedEditGroups":[]}},{"priority":31,"change":{"message":"Convert to ConsumerStatefulWidget","edits":[{"fileStamp":0,"edits":[{"offset":2424,"length":35,"replacement":"ConsumerState<HookConsumerStateful>"},{"offset":2291,"length":35,"replacement":"ConsumerState<HookConsumerStateful>"},{"offset":2204,"length":26,"replacement":"ConsumerStatefulWidget"}]}],"linkedEditGroups":[]}},{"priority":31,"change":{"message":"Convert to ConsumerStatefulWidget","edits":[{"fileStamp":0,"edits":[{"offset":2676,"length":23,"replacement":""},{"offset":2626,"length":0,"replacement":"@override\n  ConsumerState<Consumer> createState() => _ConsumerState();\n}\n\nclass _ConsumerState extends ConsumerState<Consumer> {\n"},{"offset":2575,"length":14,"replacement":"ConsumerStatefulWidget"}]}],"linkedEditGroups":[]}}]