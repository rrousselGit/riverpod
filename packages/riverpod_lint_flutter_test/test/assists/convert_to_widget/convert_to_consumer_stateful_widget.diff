Message: `Convert to ConsumerStatefulWidget`
Priority: 31
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:5`:
```
import 'package:flutter_hooks/flutter_hooks.dart';

- class Stateless extends StatelessWidget {
-   const Stateless({super.key});
- 
-   @override
+ class Stateless extends ConsumerStatefulWidget {
+   const Stateless({super.key});
+ 
+   @override
+   ConsumerState<Stateless> createState() => _StatelessState();
+ }
+ 
+ class _StatelessState extends ConsumerState<Stateless> {
@override
  Widget build(BuildContext context) {
```
---
Message: `Convert to ConsumerStatefulWidget`
Priority: 31
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:14`:
```
}

- class StatelessWithComma extends StatelessWidget {
-   const StatelessWithComma({super.key});
- 
-   @override
+ class StatelessWithComma extends ConsumerStatefulWidget {
+   const StatelessWithComma({super.key});
+ 
+   @override
+   ConsumerState<StatelessWithComma> createState() => _StatelessWithCommaState();
+ }
+ 
+ class _StatelessWithCommaState extends ConsumerState<StatelessWithComma> {
@override
  Widget build(
```
---
Message: `Convert to ConsumerStatefulWidget`
Priority: 31
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:25`:
```
}

- class Hook extends HookWidget {
-   const Hook({super.key});
- 
-   @override
+ class Hook extends ConsumerStatefulWidget {
+   const Hook({super.key});
+ 
+   @override
+   ConsumerState<Hook> createState() => _HookState();
+ }
+ 
+ class _HookState extends ConsumerState<Hook> {
@override
  Widget build(BuildContext context) {
```
---
Message: `Convert to ConsumerStatefulWidget`
Priority: 31
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:34`:
```
}

- class HookConsumer extends HookConsumerWidget {
-   const HookConsumer({super.key});
- 
-   @override
-   Widget build(
-     BuildContext context,
-     WidgetRef ref,
-   ) {
+ class HookConsumer extends ConsumerStatefulWidget {
+   const HookConsumer({super.key});
+ 
+   @override
+   ConsumerState<HookConsumer> createState() => _HookConsumerState();
+ }
+ 
+ class _HookConsumerState extends ConsumerState<HookConsumer> {
+ @override
+   Widget build(
+     BuildContext context) {
    return const Placeholder();
  }
```
---
Message: `Convert to ConsumerStatefulWidget`
Priority: 27
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:46`:
```
}

- class Stateful extends StatefulWidget {
-   const Stateful({super.key});
- 
-   @override
-   State<Stateful> createState() => _StatefulState();
- }
- 
- class _StatefulState extends State<Stateful> {
+ class Stateful extends ConsumerStatefulWidget {
+   const Stateful({super.key});
+ 
+   @override
+   ConsumerState<Stateful> createState() => _StatefulState();
+ }
+ 
+ class _StatefulState extends ConsumerState<Stateful> {
  /// Hello world
  @override
```
---
Message: `Convert to ConsumerStatefulWidget`
Priority: 27
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:62`:
```
}

- class ExplicitCreateState extends StatefulWidget {
-   const ExplicitCreateState({super.key});
- 
-   @override
-   ExplicitCreateStateState createState() => ExplicitCreateStateState();
- }
- 
- class ExplicitCreateStateState extends State<ExplicitCreateState> {
+ class ExplicitCreateState extends ConsumerStatefulWidget {
+   const ExplicitCreateState({super.key});
+ 
+   @override
+   ExplicitCreateStateState createState() => ExplicitCreateStateState();
+ }
+ 
+ class ExplicitCreateStateState extends ConsumerState<ExplicitCreateState> {
  @override
  Widget build(
```
---
Message: `Convert to ConsumerStatefulWidget`
Priority: 31
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:78`:
```
}

- class HookStateful extends StatefulHookWidget {
-   const HookStateful({super.key});
- 
-   @override
-   State<HookStateful> createState() => HookStatefulState();
- }
- 
- class HookStatefulState extends State<HookStateful> {
+ class HookStateful extends ConsumerStatefulWidget {
+   const HookStateful({super.key});
+ 
+   @override
+   ConsumerState<HookStateful> createState() => HookStatefulState();
+ }
+ 
+ class HookStatefulState extends ConsumerState<HookStateful> {
  @override
  Widget build(BuildContext context) {
```
---
Message: `Convert to ConsumerStatefulWidget`
Priority: 31
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:106`:
```
}

- class HookConsumerStateful extends StatefulHookConsumerWidget {
+ class HookConsumerStateful extends ConsumerStatefulWidget {
  const HookConsumerStateful({super.key});

```
---
Message: `Convert to ConsumerStatefulWidget`
Priority: 31
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:121`:
```
}

- class Consumer extends ConsumerWidget {
-   const Consumer({super.key});
- 
-   @override
-   Widget build(
-     BuildContext context,
-     WidgetRef ref,
-   ) {
+ class Consumer extends ConsumerStatefulWidget {
+   const Consumer({super.key});
+ 
+   @override
+   ConsumerState<Consumer> createState() => _ConsumerState();
+ }
+ 
+ class _ConsumerState extends ConsumerState<Consumer> {
+ @override
+   Widget build(
+     BuildContext context) {
    return const Placeholder();
  }
```
---
Message: `Convert to ConsumerStatefulWidget`
Priority: 31
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:133`:
```
}

- class StatelessWithField extends StatelessWidget {
-   const StatelessWithField({
-     super.key,
-     required this.field,
-   });
- 
-   final int field;
-   static final int staticField = 42;
- 
-   @override
-   Widget build(BuildContext context) {
-     return Column(
-       children: [
-         Text('$field'),
-         Text('$staticField'),
+ class StatelessWithField extends ConsumerStatefulWidget {
+   const StatelessWithField({
+     super.key,
+     required this.field,
+   });
+ 
+   final int field;
+   static final int staticField = 42;
+ 
+   @override
+   ConsumerState<StatelessWithField> createState() => _StatelessWithFieldState();
+ }
+ 
+ class _StatelessWithFieldState extends ConsumerState<StatelessWithField> {
+ @override
+   Widget build(BuildContext context) {
+     return Column(
+       children: [
+         Text('${widget.field}'),
+         Text('${StatelessWithField.staticField}'),
      ],
    );
```
---
Message: `Convert to ConsumerStatefulWidget`
Priority: 31
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:153`:
```
}

- class HookConsumerWithField extends HookConsumerWidget {
-   const HookConsumerWithField({
-     super.key,
-     required this.field,
-   });
- 
-   final int field;
-   static final int staticField = 42;
- 
-   @override
-   Widget build(
-     BuildContext context,
-     WidgetRef ref,
-   ) {
-     return Column(
-       children: [
-         Text('$field'),
-         Text('$staticField'),
+ class HookConsumerWithField extends ConsumerStatefulWidget {
+   const HookConsumerWithField({
+     super.key,
+     required this.field,
+   });
+ 
+   final int field;
+   static final int staticField = 42;
+ 
+   @override
+   ConsumerState<HookConsumerWithField> createState() => _HookConsumerWithFieldState();
+ }
+ 
+ class _HookConsumerWithFieldState extends ConsumerState<HookConsumerWithField> {
+ @override
+   Widget build(
+     BuildContext context) {
+     return Column(
+       children: [
+         Text('${widget.field}'),
+         Text('${HookConsumerWithField.staticField}'),
      ],
    );
```
---
