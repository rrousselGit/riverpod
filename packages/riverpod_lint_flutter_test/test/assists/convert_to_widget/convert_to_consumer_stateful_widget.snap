Message: `Convert to ConsumerStatefulWidget`
<<<< start: 168 -- end: 220
StatelessWidget {
  const Stateless({super.key});

 
==== start: 168 -- end: 360
ConsumerStatefulWidget {
  const Stateless({super.key});

  @override
  ConsumerState<Stateless> createState() => _StatelessState();
}

class _StatelessState extends ConsumerState<Stateless> {
>>>>
Message: `Convert to ConsumerStatefulWidget`
<<<< start: 342 -- end: 403
StatelessWidget {
  const StatelessWithComma({super.key});

 
==== start: 342 -- end: 579
ConsumerStatefulWidget {
  const StatelessWithComma({super.key});

  @override
  ConsumerState<StatelessWithComma> createState() => _StatelessWithCommaState();
}

class _StatelessWithCommaState extends ConsumerState<StatelessWithComma> {
>>>>
Message: `Convert to ConsumerStatefulWidget`
<<<< start: 520 -- end: 562
HookWidget {
  const Hook({super.key});

 
==== start: 520 -- end: 687
ConsumerStatefulWidget {
  const Hook({super.key});

  @override
  ConsumerState<Hook> createState() => _HookState();
}

class _HookState extends ConsumerState<Hook> {
>>>>
Message: `Convert to ConsumerStatefulWidget`
<<<< start: 678 -- end: 809
HookConsumerWidget {
  const HookConsumer({super.key});

  @override
  Widget build(
    BuildContext context,
    WidgetRef ref,
 
==== start: 678 -- end: 935
ConsumerStatefulWidget {
  const HookConsumer({super.key});

  @override
  ConsumerState<HookConsumer> createState() => _HookConsumerState();
}

class _HookConsumerState extends ConsumerState<HookConsumer> {
@override
  Widget build(
    BuildContext contex
>>>>
Message: `Convert to ConsumerStatefulWidget`
<<<< start: 876 -- end: 1021
StatefulWidget {
  const Stateful({super.key});

  @override
  State<Stateful> createState() => _StatefulState();
}

class _StatefulState extends
==== start: 876 -- end: 1045
ConsumerStatefulWidget {
  const Stateful({super.key});

  @override
  ConsumerState<Stateful> createState() => _StatefulState();
}

class _StatefulState extends Consume
>>>>
Message: `Convert to ConsumerStatefulWidget`
<<<< start: 1203 -- end: 1388
StatefulWidget {
  const ExplicitCreateState({super.key});

  @override
  ExplicitCreateStateState createState() => ExplicitCreateStateState();
}

class ExplicitCreateStateState extends
==== start: 1203 -- end: 1404
ConsumerStatefulWidget {
  const ExplicitCreateState({super.key});

  @override
  ExplicitCreateStateState createState() => ExplicitCreateStateState();
}

class ExplicitCreateStateState extends Consume
>>>>
Message: `Convert to ConsumerStatefulWidget`
<<<< start: 1544 -- end: 1707
StatefulHookWidget {
  const HookStateful({super.key});

  @override
  State<HookStateful> createState() => HookStatefulState();
}

class HookStatefulState extends
==== start: 1544 -- end: 1727
ConsumerStatefulWidget {
  const HookStateful({super.key});

  @override
  ConsumerState<HookStateful> createState() => HookStatefulState();
}

class HookStatefulState extends Consume
>>>>
Message: `Convert to ConsumerStatefulWidget`
<<<< start: 2204 -- end: 2223
StatefulHookConsume
==== start: 2204 -- end: 2219
ConsumerStatefu
>>>>
Message: `Convert to ConsumerStatefulWidget`
<<<< start: 2583 -- end: 2698
Widget {
  const Consumer({super.key});

  @override
  Widget build(
    BuildContext context,
    WidgetRef ref,
 
==== start: 2583 -- end: 2812
StatefulWidget {
  const Consumer({super.key});

  @override
  ConsumerState<Consumer> createState() => _ConsumerState();
}

class _ConsumerState extends ConsumerState<Consumer> {
@override
  Widget build(
    BuildContext contex
>>>>
